src/pages/EditRequestPage.js

```diff
@@ -0,0 +1,145 @@
+        /** Params from the route */
+        params: PropTypes.shape({
+            /** Which field we are editing */
+            field: PropTypes.string,
+
+            /** This is going to be the reportID for the "transaction thread" */
+            threadReportID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    /** The report object for the thread report */
+    report: reportPropTypes,
+};
+
+const defaultProps = {
+    report: {},
+};
+
+/**
+ * @param {string} fieldName
+ * @returns {string}
+ */
+function getTranslationForField(fieldName) {
+    switch (fieldName) {
+        case CONST.EDIT_REQUEST_FIELD.AMOUNT: {
+            return 'iou.amount';
+        }
+        case CONST.EDIT_REQUEST_FIELD.DESCRIPTION: {
+            return 'common.description';
+        }
+        case CONST.EDIT_REQUEST_FIELD.DATE: {
+            return 'common.date';
+        }
+        default: {
+            throw new Error('Invalid fieldName for this view');
+        }
+    }
+}
+
+function EditRequestPage(props) {
+    const parentReportAction = ReportActionsUtils.getParentReportAction(props.report);
+    const moneyRequestReportAction = ReportUtils.getMoneyRequestAction(parentReportAction);
+    const transactionAmount = moneyRequestReportAction.amount;
+    const transactionCurrency = moneyRequestReportAction.currency;
+    const transactionDescription = moneyRequestReportAction.comment;
+    const created = parentReportAction.created;
+    const threadReportID = lodashGet(props, ['route', 'params', 'threadReportID'], '');
+    const field = lodashGet(props, ['route', 'params', 'field'], '');
+
+    function updateTransactionWithChanges(changes) {
+        // Update the transaction...
+        // eslint-disable-next-line no-console
+        console.log({changes});
+
+        // Note: The "modal" we are dismissing is the MoneyRequestAmountPage
+        Navigation.dismissModal();
+    }
+
+    if (field === CONST.EDIT_REQUEST_FIELD.DESCRIPTION) {
+        return (
+            <RequestDescription
+                onBackButtonPress={() => Navigation.goBack()}
+                validate={() => ({})}
+                submit={({comment}) => updateTransactionWithChanges({comment})}
+                headerTitle={props.translate('common.description')}
+                formID={ONYXKEYS.FORMS.MONEY_REQUEST_DESCRIPTION_FORM}
+                textInputID="comment"
+                textInputLabel={props.translate('moneyRequestConfirmationList.whatsItFor')}
+                textInputDefaultValue={transactionDescription}
+            />
+        );
+    }
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={field === CONST.EDIT_REQUEST_FIELD.AMOUNT}>
+            <HeaderWithBackButton
+                title={props.translate(getTranslationForField(field))}
+                shouldShowBackButton
+                onBackButtonPress={() => Navigation.dismissModal()}
+                onCloseButtonPress={() => Navigation.dismissModal()}
```

Comment:
I don't think we show the Close button on this page, so we can probable remove this.

Reply:
Yeah I think the close button... doesn't exist anymore... ?

Reply:
Rather, we do but only is `shouldShowCloseButton` is passed